core/src/main/java/io/kestra/core/models/tasks/metrics/CounterMetric.java

```diff
@@ -0,0 +1,49 @@
+package io.kestra.core.models.tasks.metrics;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.models.executions.AbstractMetricEntry;
+import io.kestra.core.models.executions.metrics.Counter;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.runners.RunContext;
+import jakarta.validation.constraints.NotNull;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@AllArgsConstructor
+@SuperBuilder
+public class CounterMetric extends AbstractMetric {
+    public static final String TYPE = "counter";
+
+    @NotNull
+    @JsonInclude
+    private final String type = TYPE;
```

Comment:
type should be in the abstract with no default as the user must specify the type so the deserialization can select the correct impl